src/internal/connector/onedrive/collection.go

```diff
@@ -72,8 +72,11 @@ type Collection struct {
 	folderPath path.Path
 	// M365 IDs of file items within this collection
 	driveItems map[string]models.DriveItemable
-	// M365 ID of the drive this collection was created from
-	driveID        string
+
+	// Primary M365 ID of the drive this collection was created from
+	driveID string
+	// Display Name of the associated drive
+	driveName      string
```

Comment:
quick check, are the driveName and driveID getting added to the clues ctx anywhere?  Quick search on my current branch doesn't turn up anything.  We should probably add those somewhere.  Might come in handy for future debugging.